### PR TITLE
Fix clang compiler warnings

### PR DESCRIPTION
**Pull request by diyessi**

### CODE REVIEW

File: src/ngraph/op/util/unary_elementwise_arithmetic.cpp [hunk 1]
```diff
@@ -18,6 +18,16 @@
 
 using namespace ngraph;
 
+op::util::UnaryElementwiseArithmetic::UnaryElementwiseArithmetic()
```
Comment (aprocter) [inline src/ngraph/op/util/unary_elementwise_arithmetic.cpp]: I think this is extraneous to this PR, right?

File: src/tools/nbench/nbench.cpp [hunk 2]
```diff
@@ -85,7 +85,7 @@ multimap<size_t, string> aggregate_timing_details(const vector<PerfShape>& perf_
     }
 
     multimap<size_t, string> rc;
-    for (const pair<string, size_t>& t : timing)
+    for (auto& t : timing)
```
Comment (rkimballn1) [inline src/tools/nbench/nbench.cpp]: Can this be `const auto&`? I much prefer to keep the const

Comment (diyessi) [reply src/tools/nbench/nbench.cpp]: `auto` can't make `const`ness go away.

File: src/ngraph/op/util/unary_elementwise_arithmetic.cpp (hunk 1 above)
Comment (diyessi) [reply src/ngraph/op/util/unary_elementwise_arithmetic.cpp]: Gone.

File: src/ngraph/op/util/unary_elementwise_arithmetic.hpp [hunk 3]
```diff
@@ -44,6 +44,13 @@ namespace ngraph
             class UnaryElementwiseArithmetic : public Op
             {
             protected:
+                UnaryElementwiseArithmetic();
```
Comment (aprocter) [inline src/ngraph/op/util/unary_elementwise_arithmetic.hpp]: I think this is another one that snuck in- sorry, should have flagged it on last review.

Comment (diyessi) [reply src/ngraph/op/util/unary_elementwise_arithmetic.hpp]: Evicted. I don't see any more.

**Pull request merged**